From a3f2f8b2c9c1b646a908d0d366dd12fd1d401471 Mon Sep 17 00:00:00 2001
From: Armand Grillet <armand.grillet@outlook.com>
Date: Wed, 28 Feb 2018 21:00:24 +0100
Subject: [PATCH 3/4] Mesos UI: updated the URL generation to make it work with
 adminrouter.

- Use /mesos as the leading master URL prefix.
- Use /mesos/agent/{agent_id}/{process_name} as the agent URL prefix.
- Redirect from Mesos webui location (host:5050) to adminrouter
  (host/mesos).
---
 src/webui/app/controllers.js | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/src/webui/app/controllers.js b/src/webui/app/controllers.js
index 8049cf611..74f65f574 100644
--- a/src/webui/app/controllers.js
+++ b/src/webui/app/controllers.js
@@ -51,10 +51,10 @@
   // (e.g., hosting '/slave/log' for each agent log, we don't
   // namespace metrics within '/metrics/snapshot', etc).
   function agentURLPrefix(agent, includeProcessId) {
-    var port = agent.pid.substring(agent.pid.lastIndexOf(':') + 1);
     var processId = agent.pid.substring(0, agent.pid.indexOf('@'));
 
-    var url = '//' + agent.hostname + ':' + port;
+    // Admin Router routes '/agent/<id>' to the agent.
+    var url = '/agent/' + agent.id;
 
     if (includeProcessId) {
       url += '/' + processId;
@@ -63,14 +63,9 @@
     return url;
   }
 
-  function leadingMasterURLPrefix(leader_info) {
-    if (leader_info) {
-      return '//' + leader_info.hostname + ':' + leader_info.port;
-    }
-
-    // If we do not have `leader_info` available (e.g. the first
-    // time we are retrieving state), fallback to the current master.
-    return '';
+  function leadingMasterURLPrefix(_leader_info) {
+    // Admin Router routes '/mesos' to the leading master.
+    return '/mesos';
   }
 
   // Invokes the pailer, building the endpoint URL with the specified urlPrefix
@@ -382,6 +377,12 @@
   mesosApp.controller('MainCtrl', [
       '$scope', '$http', '$location', '$timeout', '$modal',
       function($scope, $http, $location, $timeout, $modal) {
+    // Redirect from the direct Mesos webui URL to the DC/OS URL that goes through adminrouter.
+    if ($location.port() === 5050) {
+      var url = $location.protocol() + "://" + $location.host() + "/mesos";
+      window.location = url;
+    }
+
     $scope.doneLoading = true;
 
     // Adding bindings into scope so that they can be used from within
-- 
2.17.0

